pkg/providers/instancetype/instancetypes.go

```diff
@@ -267,6 +271,24 @@ func (p *DefaultProvider) isInstanceTypeSupportedByImageFamily(skuName, imageFam
 	}
 }
 
+func (p *DefaultProvider) isInstanceTypeSupportedByEncryptionAtHost(sku *skewer.SKU, nodeClass *v1beta1.AKSNodeClass) bool {
+	// If EncryptionAtHost is not enabled in the nodeclass, all instance types are supported
+	if !lo.FromPtr(nodeClass.Spec.EncryptionAtHost) {
+		return true
+	}
+	// If EncryptionAtHost is enabled, only include instance types that support it
+	return p.supportsEncryptionAtHost(sku)
+}
+
+// supportsEncryptionAtHost checks if the SKU supports encryption at host
+func (p *DefaultProvider) supportsEncryptionAtHost(sku *skewer.SKU) bool {
+	value, err := sku.GetCapabilityString("EncryptionAtHostSupported")
+	if err != nil {
+		return false
+	}
+	return value == "True"
```

Comment:
Is it always cased this way?

Should we be case-insensitive just in case?

Reply:
We could use strings.EqualFold just in case, I did a quick CLI List + Grep but only found UpperCase true for this field though. 


Probably good to code defensively.